test/integration/agent_hosting_deploy_test.go

```diff
@@ -217,6 +217,26 @@ var _ = ginkgo.Describe("Agent deploy", func() {
 			return nil
 		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
 
+		// do nothing if cluster is deleting and addon is not deleted
+		cluster, err := hubClusterClient.ClusterV1().ManagedClusters().Get(context.Background(), managedClusterName, metav1.GetOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		cluster.SetFinalizers([]string{"cluster.open-cluster-management.io/api-resource-cleanup"})
+		_, err = hubClusterClient.ClusterV1().ManagedClusters().Update(context.Background(), cluster, metav1.UpdateOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		err = hubClusterClient.ClusterV1().ManagedClusters().Delete(context.Background(), managedClusterName, metav1.DeleteOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		gomega.Eventually(func() error {
+			time.Sleep(5 * time.Second) // wait 5 seconds to sync
+			work, err = hubWorkClient.WorkV1().ManifestWorks(hostingClusterName).Get(context.Background(),
+				hostingManifestWorkName, metav1.GetOptions{})
+			if err != nil {
+				gomega.Expect(errors.IsNotFound(err)).ToNot(gomega.BeTrue())
+				return err
```

Comment:
I don't quite understand here, will it return err even if it is a not found err?

Reply:
fixed